app/services/workflows/non_passported_workflow.rb

```diff
@@ -17,8 +18,11 @@ def collate_and_assess_gross_income
       Assessors::GrossIncomeAssessor.call(assessment)
     end
 
-    def disposable_income_assessment
+    def collate_outgoings
       Collators::OutgoingsCollator.call(assessment)
+    end
+
+    def disposable_income_assessment
       Collators::DisposableIncomeCollator.call(assessment)
```

Comment:
Should this `Collators::DisposableIncomeCollator.call(assessment) ` be moved out into a separate method. This current disposable_income_assessment is called only when the  `gross_income_summary.eligible?` but you mention you still need to do the DisposableIncomeCollator, which is blocked by this  ` if gross_income_summary.eligible?` on line 9